### DISCUSSION TITLE
How to match Tatool Data with Survey Data

### DISCUSSION THREAD

**Discussion by mayamuh2** [tatool/tatool-web#151] [Q&A] [answered]
Hi, so we are using 4 tasks from Tatool for our study. After they complete the tasks, we want them to complete a demographics survey. We used the redirect url feature on the general settings, but then there is no way to identify which tatool responses match which survey responses. What is your best advice for how to match data from our survey to data from the tatool tasks?

**Comment by alocher** [comment]
Currently there is no link between the two so you would have to for example include a question in your survey where the participant enters the same unique External ID you used as part of the Tatool Module URL. 

I think we could improve this by adding a feature to pass the extid automatically via the URL as query parameter. For example in the case of Qualtrics you would then be able to save this in your survey data using the [Embedded Data](https://www.qualtrics.com/support/survey-platform/survey-module/survey-flow/standard-elements/embedded-data/#SettingValuesFromTheSurveyURL) as far as I understand.

Let us know if this would be a useful feature and we can try and implement this on tatool-web.com.

**Comment by alocher** [comment, marked as answer]
This feature has now been added. If you provide a "Redirect URL" for your module, Tatool will automatically append the unique "sessiontoken" and and "extid" if used in your public module URL. Hope this will be useful.
